digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/ProhibitionService.scala

```diff
@@ -0,0 +1,191 @@
+    }
+  }
+
+  override protected def updateProjected(toUpdate: Seq[PersistedLinearAsset], persisted: Map[Long, Seq[PersistedLinearAsset]]) = {
+    def valueChanged(assetToPersist: PersistedLinearAsset, persistedLinearAsset: Option[PersistedLinearAsset]) = {
+      !persistedLinearAsset.exists(_.value == assetToPersist.value)
+    }
+    def mValueChanged(assetToPersist: PersistedLinearAsset, persistedLinearAsset: Option[PersistedLinearAsset]) = {
+      !persistedLinearAsset.exists(pl => pl.startMeasure == assetToPersist.startMeasure &&
+        pl.endMeasure == assetToPersist.endMeasure &&
+        pl.vvhTimeStamp == assetToPersist.vvhTimeStamp)
+    }
+    def sideCodeChanged(assetToPersist: PersistedLinearAsset, persistedLinearAsset: Option[PersistedLinearAsset]) = {
+      !persistedLinearAsset.exists(_.sideCode == assetToPersist.sideCode)
+    }
+    toUpdate.foreach { linearAsset =>
+      val persistedLinearAsset = persisted.getOrElse(linearAsset.id, Seq()).headOption
+      val id = linearAsset.id
+      if (valueChanged(linearAsset, persistedLinearAsset)) {
+        linearAsset.value match {
+          case Some(prohibitions: Prohibitions) =>
+            dao.updateProhibitionValue(id, prohibitions, LinearAssetTypes.VvhGenerated)
+          case _ => None
+        }
+      }
+      if (mValueChanged(linearAsset, persistedLinearAsset)) dao.updateMValues(linearAsset.id, (linearAsset.startMeasure, linearAsset.endMeasure), linearAsset.vvhTimeStamp)
+      if (sideCodeChanged(linearAsset, persistedLinearAsset)) dao.updateSideCode(linearAsset.id, SideCode(linearAsset.sideCode))
+    }
+  }
+
+  override protected def updateWithoutTransaction(ids: Seq[Long], value: Value, username: String, measures: Option[Measures] = None, vvhTimeStamp: Option[Long] = None, sideCode: Option[Int] = None): Seq[Long] = {
+    if (ids.isEmpty)
+      return ids
+
+    val assetTypeId = sql"""select ID, ASSET_TYPE_ID from ASSET where ID in (#${ids.mkString(",")})""".as[(Long, Int)].list
+    val assetTypeById = assetTypeId.foldLeft(Map.empty[Long, Int]) { case (m, (id, typeId)) => m + (id -> typeId)}
+
+    ids.flatMap { id =>
+      val typeId = assetTypeById(id)
+      value match {
+        case prohibitions: Prohibitions =>
+          updateValueByExpiration(id, prohibitions, typeId, username, measures, vvhTimeStamp, sideCode)
+        case _ =>
+          Some(id)
+      }
+    }
+  }
+
+  protected def updateValueByExpiration(assetId: Long, prohibitions: Prohibitions, assetTypeId: Int, username: String, measures: Option[Measures], vvhTimeStamp: Option[Long], sideCode: Option[Int]): Option[Long] = {
+    //Get Old Asset
+    dao.fetchProhibitionsByIds(assetTypeId, Set(assetId)).headOption.map {
+      oldAsset =>
+      if(oldAsset.value.contains(prohibitions)){
+        dao.updateProhibitionValue(assetId, prohibitions, username, measures).head
+      } else {
+        //Expire the old asset
+        dao.updateExpiration(assetId)
```

Comment:
Maybe this new method should be replaced by dao.updateExpiration(assetId, expired = true, username)